src/ProjectOrigin.WalletSystem.Server/Models/CertificateState.cs

```diff
@@ -0,0 +1,20 @@
+namespace ProjectOrigin.WalletSystem.Server.Models;
+
+public enum CertificateState
+{
+    /// <summary>
+    /// The certificate could not be loaded or is invalid.
+    /// other -1xx codes can be added for different invalid states.
+    /// </summary>
+    Invalid = -100,
+
+    /// <summary>
+    /// The certificate has been inserted, but header information has not yet been loaded from the registry.
+    /// </summary>
+    Inserted = 0,
+
+    /// <summary>
+    /// The certificate has been loaded from the registry and is valid.
+    /// </summary>
+    Valid = 100,
```

Comment:
Is it needed to have ranges on the State? It is not exposed through the gRPC interface